### ISSUE TITLE
[Android] Ensure Noto-Serif is properly applied to Headings

### ISSUE THREAD

**Issue by iamthomasbishop** [wordpress-mobile/gutenberg-mobile#1147]
Something that's been bugging me for a while: On Android, it doesn't look like Noto Serif is properly applied as the font-family on Headings. Side-by-side for comparison:

![image](https://user-images.githubusercontent.com/1200432/59632626-3bc85600-9110-11e9-94aa-7e5cb4bcc2f6.png)

// cc @pinarol I know we did some work on headings and weights, but not sure if that is relevant to this.

**Comment by maxme** [comment]
I did a quick test in Photoshop to check if this was the correct font. To my eyes, this is Noto Serif font in this heading screenshot, the vertical space between characters seems smaller than in the iOS screenshot. I think the iOS screenshot shows the bold version of Noto, while it's the regular one in Android. 

Here is a screenshot of the test I did in Photoshop (note the modified VA parameter):

<img width="501" alt="Screenshot 2019-07-24 at 09 34 21" src="https://user-images.githubusercontent.com/40213/61774218-b1ea6780-adf6-11e9-8b43-62f871e3a90e.png">


I also double checked the code, and I don't see anything that could change it to another font. The default font for Android is set here:

https://github.com/wordpress-mobile/gutenberg-mobile/blob/c00696888f0eee80a377d90608e6e53bfee0be47/src/_native.android.scss#L5

Note: the `serif` font in Android correspond to `NotoSerif-Regular.ttf`

**Comment by pinarol** [comment]
On iOS side we are using Noto Serif Bold by default on Heading block as described [here](https://github.com/wordpress-mobile/gutenberg-mobile/issues/351#issuecomment-456522533),
When user taps B button we are applying a trick to make it look bolder, we are actually adding some shadow under the hood. Related PR: https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1127 We did this because it was hard to distinguish Heading from Paragraph on iOS.

**Comment by maxme** [comment]
Thanks @pinarol this explains why there is a visual difference. @iamthomasbishop let me know if you want to change something there or if we can close that one.

**Comment by iamthomasbishop** [comment]
Ah, that's interesting, so it *is* Noto Serif but the letter-spacing is really condensed. Along with what @pinarol mentioned, I think setting the letter-spacing to zero should solve it then!

**Comment by iamthomasbishop** [comment]
One more thing, just to be sure – can we set our "baseline" letter-spacing to `0` so this applies to all text-based blocks?

**Comment by maxme** [comment]
>  can we set our "baseline" letter-spacing to `0` so this applies to all text-based blocks?

I think the letter-spacing is using the default value (I can't find any reference to a specific letter-spacing config). I'm actually not sure why the headings look so condensed. I can do another test in the default Android Text component to check if something happens on our side. There might be something in Aztec or in GB->Aztec glue code that changes the letter-spacing but I haven't investigated yet.

**Comment by maxme** [comment]
I investigated a bit more and Aztec is using the default Android letter-spacing parameter which is `0`. We can override it, for example:

Unset (ie. default: 0)|letter-spacing:0.05|letter-spacing:0.1
-|-|-
![Screenshot_1564062441](https://user-images.githubusercontent.com/40213/61879601-a466d780-aef3-11e9-8cdf-407106eba748.png)| ![Screenshot_1564062426](https://user-images.githubusercontent.com/40213/61879618-ab8de580-aef3-11e9-9570-43c574a9d569.png)| ![Screenshot_1564062573](https://user-images.githubusercontent.com/40213/61879671-cf512b80-aef3-11e9-8f2f-26267505d38c.png)

Android documentation about TextView (EditText) (underlying class used by Aztec): https://developer.android.com/reference/android/widget/TextView.html#attr_android:letterSpacing

> can we set our "baseline" letter-spacing to 0 so this applies to all text-based blocks?

Yes and no.... If we add that parameter to Aztec (and the RN glue code) we can update the baseline letter-spacing for all Aztec-based blocks. But from what I found, not all text based blocks use Aztec (eg. "write caption..." param in video block use the standard TextEdit). We could _maybe_ (I need to investigate more) update that parameters for all TextEdit based blocks.

**Comment by mchowning** [comment]
>not all text based blocks use Aztec (eg. "write caption..." param in video block use the standard TextEdit).

FWIW I would assume the video block caption will be Aztec-based once we add rich text editing. AFAIK the (unreleased) code block is the only other block that is not using Aztec. I don't know what the plans are for the code block. Also, maybe we wouldn't want to change the spacing for the code block?

**Comment by iamthomasbishop** [comment]
Good point, we might not want to adjust the letter-spacing on the code block – only in these cases where the letter-spacing is off. The code block uses a monospace font so it might be fine (haven’t seen this to confirm yet).

**Comment by hypest** [comment]
Is there anything left to do on this ticket? Can we perhaps close it @iamthomasbishop ?

**Comment by iamthomasbishop** [comment]
@hypest I don't think this has been resolved on Android yet, as I'm still seeing the same weight font on Heading blocks. Here's an example where you can see the font-weight diff between the title (correct weight) and heading block text (lighter):

<a href="https://user-images.githubusercontent.com/1200432/76636352-e55d3c00-6516-11ea-90d0-0fed8fadbffa.png"><img src="https://user-images.githubusercontent.com/1200432/76636352-e55d3c00-6516-11ea-90d0-0fed8fadbffa.png" width="360"></a>
